tsdb/engine/tsm1/engine.go

```diff
@@ -2065,6 +2070,17 @@ func (e *Engine) compact(wg *sync.WaitGroup) {
 			return
 
 		case <-t.C:
+			// See if compactions are disabled.
```

Comment:
i'd think about this in terms of `lastDisableMsg` instead of `nextDisableMsg`.

this reads a bit more clearly.  instead of asking, "is now after the nextDisabledMessage" we can ask, "has it been 15 mins since `lastDisabledMsg`?"

```
  6       // See if compactions are disabled.
     5       doNotCompactFile := filepath.Join(e.Path(), DoNotCompactFile)
     4       _, err := os.Stat(doNotCompactFile)
~    3       if err == nil && time.Since(lastDisabledMsg) > time.Minute*15 {
~    2         e.logger.Info("TSM compaction disabled", logger.Shard(e.id), zap.String("reason", doNotCompactFile))
~_   1         lastDisabledMsg = time.Now()
  2079         continue
     1       }

```

or something like that.

this seems clearer, saves an indent lev

Reply:
also, i'd probably change the name of `doNotCompatFile` and `DoNotCompactFile` to a noun like, `compactionDisableFile` and `CompactionDisableFile`.

Reply:
I prefer to keep names private until they to be made public. It's easy to capitalize as part of some other PR that needs access.

Reply:
i agree.  there's already `DoNotCompactFile` though -- was trying to make minimal changes.

Reply:
this feels like bike shedding though. :)